### ISSUE TITLE
Migration from 0.25.3 to 1.0.0-rc.1: : Could not get GitRepository flux-system/flux-system

### ISSUE THREAD

**Issue by matmb-cld** [fluxcd/terraform-provider-flux#453]
Hello,

I'm trying to migrate from 0.25.3 to 1.0.0-rc.1 by following this [documentation](https://registry.terraform.io/providers/fluxcd/flux/latest/docs/guides/migrating-to-resource).
I've modifed the terraform file by deleting the deprecated ressources and the only configured flux ressource is now
```
resource "flux_bootstrap_git" "this" {
  depends_on = [gitlab_deploy_key.this]
  path = var.flux_sync_target_path
```
I 've performed   `terraform init --upgrade`
Then I've tried to import the ressource  `terraform import flux_bootstrap_git.this flux-system`

But I'm facing this error
```
│ Error: Could not get GitRepository flux-system/flux-system                                                                                                                                                       
│                                                                                                                                                                                                          
│ no matches for kind "GitRepository" in version "source.toolkit.fluxcd.io/v1"
```

However the Flux GitRepository is present
```
kubectl -n flux-system get gitrepositories.source.toolkit.fluxcd.io                                                                                    
NAME          URL                                                                          AGE    READY   STATUS
flux-system   ssh://git@sourcehub.orange-business.com/sf-clouds-externe/cce-tools-fe.git   171d   True    stored artifact for revision 'main@sha1:7386a8a9c48086d6da0fb8d8b840a9b085ea5576'
```

But the GitRepository API version is **v1beta2**. Could it be the reason of this error ?

```
k -n flux-system describe gitrepositories.source.toolkit.fluxcd.io flux-system                                                                   khsb5085@hardbop-mood 
Name:         flux-system                                                                                                                                                                                          
Namespace:    flux-system                                                                                                                                                                                          
Labels:       kustomize.toolkit.fluxcd.io/name=flux-system                                                                                                                                                         
              kustomize.toolkit.fluxcd.io/namespace=flux-system                                                                                                                                                    
Annotations:  reconcile.fluxcd.io/requestedAt: 2023-04-11T14:36:21.968679+02:00                                                                                                                                    
API Version:  source.toolkit.fluxcd.io/v1beta2                                                                                                                                                                     
Kind:         GitRepository                                                                                                                                                                                        
Metadata:                                                                                                                                                                                                          
  Creation Timestamp:  2022-11-07T09:15:36Z                                                                                                                                                                        
  Finalizers:                                                                                                                                                                                                      
    finalizers.fluxcd.io                                                                                                                                                                                           
  Generation:  1                                                                                                                                                                                                   
  Managed Fields:                                                                                                                                                                                                  
    API Version:  source.toolkit.fluxcd.io/v1beta2                                                                                                                                                                 
    Fields Type:  FieldsV1                                        
......
```

Mathieu

**Comment by phillebaba** [comment]
I would suggest that you first to the migration from datasource to resource without changing the version. Then updating the version of the provider after that has been completed. I have not tested what you are doing as it would be difficult to verify every combination.

**Comment by matmb-cld** [comment]
Since I'm not sure to  I properly understood what you are proposing, here is what i've done.
With 0.25.3 version, only add the resource flux_bootstrap_git in terraform then perform terraform apply or terraform import. But in the both cases, I get SIGSEGV crash (see attached file for details).

[apply_sigsegv.log](https://github.com/fluxcd/terraform-provider-flux/files/11352896/apply_sigsegv.log)

**Comment by phillebaba** [comment]
What I mean is that you should do the migration from datasource to resource without changing the version och the provider.

Looking at the logs I think that the issue is related to you provider configuration. Could you share how your provider looks?

**Comment by matmb-cld** [comment]
This what I don't understand: which datasource do I should migrate from 0.25.3 version since  flux_bootstrap_git resource is only available in 1.0.0-rc.1 version ?

Here is my provider configuration
```
provider "flux" {}

provider "kubectl" {
  config_path    = "~/.kube/prp"
  config_context = "prp"
}

provider "kubernetes" {
  config_path    = "~/.kube/prp"
  config_context = "prp"
}

provider "gitlab" {
  token    = var.gitlab_token
  base_url = var.gitlab_api_url
}
```

FI, what has triggered the SIGSEGV is that I only added the flux_bootstrap_git resource in the 0.25.3 terraform file.

**Comment by phillebaba** [comment]
I am working on fixing the SIGSEGV, it should throw an error instead.

The issue that you have is that you have missed part of the migration guide which includes modifying the provider configuration. You need to add the Kubernetes configuration to the Flux provider, as it is not managing Kubernetes resources for you.

```hcl
provider "flux" {
  kubernetes = {
    config_path    = "~/.kube/prp"
    config_context = "prp"
  }
}
```

**Comment by matmb-cld** [comment]
I figured out what was wrong on my end. I thought  kubernetes and git configuration was only available with 1.0.0.-rc1 version. But it is not the case :) 
Here are the steps I performed to migrate:
1. Add the git and kubernetes conf to the flux provider
2. Update terraform objects (remove the old ones and add resource "flux_bootstrap_git"
3. Import of the new resource: terraform import flux_bootstrap_git.this flux-system
4. Removal of old resources:  terraform state rm ....
5. Update of providers (flux to 1.0.0-rc.1, remove kubernetes and kubectl) then terraform init --upgrade
6. Then finally: terraform apply

A great thanks for your help.

**Comment by DArtagan** [comment]
Leaving a comment for posterity.  I also hit this error - and my version of the flux provider was old enough to not yet have `flux_bootstrap_git`.  So the approach that worked for me:
1. Upraded the provider version, going one terminal minor release at a time - `0.22.3`, `0.24.3`, `0.25.3` (probably overly cautious).  Update the version, `terraform init --upgrade`, `terraform apply`, repeat.
2. Then followed the migration guide.
3. Then upgraded to `1.0.0-rc.3`
